org.metaborg.spoofax.shell.eclipse/src/main/java/org/metaborg/spoofax/shell/client/eclipse/impl/hooks/EclipseMessageHook.java

```diff
@@ -26,13 +34,16 @@ public EclipseMessageHook(IDisplay display) {
 
     @Override
     public void accept(StyledText message) {
-        // TODO: run this in UI thread in a nice way!
-        Display.getDefault().asyncExec(new Runnable() {
+        Job job = new UIJob("REPL Message Hook") {
             @Override
-            public void run() {
+            public IStatus runInUIThread(IProgressMonitor monitor) {
```

Comment:
Sure, if we can figure out a nice way to do this. However, in #29 the hooks have been reworked so that a hook is returned to the client. This allows the client to schedule the hook however it needs, so we can schedule all hooks on the UI thread in one location -- so perhaps, we shouldn't look into resolving this here but do it in #29 instead.